Fix Preceding Child Records for Auto Account Creation

georgi-l95 opened the pull request:
**Description**:
This PR adds new method in the Handle Context class, which takes care of creating removable preceding child records. This is needed for CryptoTransfer for example towards non existing account.
Added new test in the AutoCreationSuite which covers this new case and passes in mono codebase as well.
With this change we cover this happy path from issue https://github.com/hashgraph/hedera-services/issues/9424:

- [x] A "classic transfer" sending an NFT to an absent receiver address (outside long-zero range) generates a preceding child with a synthetic CryptoCreate and TransactionRecord identical that created by mono-service when it resolves to SUCCESS.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:
<!-- Provide logs, performance numbers or screenshots of the new functionality -->

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

netopyr reviewed (changes requested):
This PR does not fit to the current design of `RecordListBuilder`. What do you want to accomplish?

georgi-l95 commented:
> This PR does not fit to the current design of `RecordListBuilder`. What do you want to accomplish?

You can checkout more in #9424, but basically some child records needs to be preceding. I've added test which check those records and passes on mono, but not here. Therefore I've added the needed changes to adhere to the needed business logic, so that the same tests case can pass on modularization as well.
